src/newssourceComponent.js

```diff
@@ -0,0 +1,29 @@
+import errorPopupHandler from "./errorHandler.js";
+
+export default class ArticlesFetcher {
```

Comment:
here you violate the single responsibility principle - you have a class that is named `...Fetcher`, actually fetches data but also (unexpectedly) renders articles.